source/perflib/basename.c

```diff
@@ -7,26 +7,42 @@
  * https://www.openssl.org/source/license.html
  */
 
-#ifdef	_WIN32
-
 #include <string.h>
 
+#include <openssl/crypto.h>
+
 /*
  * windows variant of basename(3). works on ASCIIZ only.
  * simple and perhaps naive implementation too.
  */
-const char * basename(const char *path)
+const char *basename(const char *path)
 {
-	const char *rv;
+	const char *rv, *tmp;
+	const char *dirnamesep;
+	size_t dirseplen;
+
+	dirnamesep = OPENSSL_info(OPENSSL_INFO_DIR_FILENAME_SEPARATOR);
+	if (dirnamesep == NULL)
+		return (NULL);
 
-	rv = (const char *)strrchr(path, '\\');
-	if (rv != NULL) {
-		rv++;
+	dirseplen = strlen(dirnamesep);
+	if (dirseplen == 1) {
+
+		rv = (const char *)strrchr(path, *dirnamesep);
+		if (rv != NULL) {
+			rv++;
+			if (*rv == '\0')
+				rv = path;
```

Comment:
These two lines are the same as 43-44 - would it be better to just have this once, outside the `if-else`?